pkg/component/etcd/etcd/crd.go

```diff
@@ -104,3 +110,13 @@ func (c *crd) Destroy(ctx context.Context) error {
 
 	return flow.Parallel(fns...)(ctx)
 }
+
+// Wait signals whether a CRD is ready or needs more time to be deployed.
+func (c *crd) Wait(ctx context.Context) error {
+	return kubernetesutils.WaitUntilCRDManifestsReady(ctx, c.client, maps.Keys(crdNameToManifest))
+}
+
+// WaitCleanup for destruction to finish and component to be fully removed. crdDeployer does not need to wait for cleanup.
+func (c *crd) WaitCleanup(_ context.Context) error {
+	return nil
```

Comment:
Why not implemented? You could similarly extract the `WaitCleanup` function from the `CRDDeployer` and call it here.

Reply:
Wouldn't it even better to use the `CRDDeployer`?